Alicerce/Sources/Persistence/DiskMemoryPersistence.swift

```diff
@@ -0,0 +1,326 @@
+
+            let fileSizes = urls.map {
+                UInt64($0.resourceValues(forKeys: [.fileSizeKey]).fileSize ?? 0)
+            }
+
+            self.usedDiskSize = fileSizes.reduce(UInt64(0), +)
+        }
+
+        readOperationQueue.addOperation(calculateUsedSizeOperation)
+    }
+
+    private func diskPath(for key: Persistence.Key) -> String {
+        return "\(configuration.path)/\(key)"
+    }
+
+    private func hasDiskCacheDirectory() -> Bool {
+        return fileManager.fileExists(atPath: configuration.path)
+    }
+
+    // FIX: Should this error be propagated to the top level?
+    private func createDiskCacheDirectory() -> Bool {
+        do {
+            try fileManager.createDirectory(atPath: configuration.path, withIntermediateDirectories: true, attributes: nil)
+        }
+        catch let error {
+            assertionFailure("failed to create directory `\(configuration.path)` with error: \n\(error)")
+            return false
+        }
+
+        return true
+    }
+
+    private func directoryContents(with keys: [URLResourceKey]) -> [URL] {
+        guard let url = URL(string: configuration.path) else {
+            assertionFailure("💥 failed to create url with path 👉 `\(configuration.path)`")
+            return []
+        }
+
+        guard let urls = try? fileManager.contentsOfDirectory(at: url,
+                                                              includingPropertiesForKeys: keys,
+                                                              options: .skipsPackageDescendants)
+            else {
+                assertionFailure("💥 failed to read directory content for path 👉 \(url.absoluteString)")
+                return []
+        }
+
+        return urls
+    }
+
+    // MARK: - Operations
+
+    private func createEvictOperation() -> DiskMemoryBlockOperation {
+        return DiskMemoryBlockOperation(qos: .background) { [unowned self] in
+
+            // Check if should run eviction
+            guard self.configuration.diskLimit < self.usedDiskSize else {
+                return
+            }
+
+            let extraOccupiedDiskSize = self.usedDiskSize - self.configuration.diskLimit
+
+            let urls = self.directoryContents(with: [.contentAccessDateKey, .fileSizeKey])
+
+            typealias FileAccessTimeSizeTuple = (accessTime: TimeInterval, size: UInt64)
+            typealias FilePathAttributesTuple = (path: String, fileAttr: FileAccessTimeSizeTuple)
+
+            let accessTimes: [FilePathAttributesTuple] = urls.map {
+                let resourceValue = $0.resourceValues(forKeys: [.contentAccessDateKey, .fileSizeKey])
+
+                return ($0.path, (resourceValue.contentAccessDate?.timeIntervalSince1970 ?? 0, UInt64(resourceValue.fileSize ?? 0)))
+                }.sorted { $0.fileAttr.accessTime > $1.fileAttr.accessTime }
+
+            var evictSize: UInt64 = 0
+
+            let filesToRemove = accessTimes.filter {
```

Comment:
It somehow feels strange to `filter` the sequence while at the same time incrementing a counter.

Perhaps a more appropriate operator would be the `prefix(while:)`:

```swift
let filesToRemove = accessTimes.prefix { 
    guard evictSize < extraOccupiedDiskSize else { return false }
    evictSize += $0.fileAttr.size
    return true
}
```

The main difference (besides the name) is that this returns an `ArraySlice` instead of an `Array`.

It's just 💄 anyway, so feel free to ignore 😆